packages/bull-vault/src/FlashBull.sol

```diff
@@ -144,10 +135,14 @@ contract FlashBull is UniBull {
             abi.encodePacked(_ethToCrab)
         );
 
-        (uint256 ethInCrab, uint256 squeethInCrab) = IBullStrategy(bullStrategy).getCrabVaultDetails(); 
-        // USDC we pay to flashswap
-
-        (ethToLend, usdcToBorrow) = IBullStrategy(bullStrategy).calcLeverageEthUsdc(crabAmount, share, ethInCrab, squeethInCrab, IERC20(crab).totalSupply());
+        (ethInCrab, squeethInCrab) = IBullStrategy(bullStrategy).getCrabVaultDetails(); 
+        uint256 share;
+        if (IERC20(bullStrategy).totalSupply() == 0) {
```

Comment:
I feel like we have some repeated code like this - maybe consider if we can refactor to function or something else to make it cleaner?